python-recipes/agents/00_langgraph_redis_agentic_rag.ipynb

```diff
@@ -0,0 +1,696 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "id": "1VK8QKOVG2Ek",
+      "metadata": {
+        "id": "1VK8QKOVG2Ek"
+      },
+      "source": [
+        "![Redis](https://redis.io/wp-content/uploads/2024/04/Logotype.svg?auto=webp&quality=85,75&width=120)\n",
+        "\n",
+        "# Agentic RAG with LangGraph and Redis\n",
+        "\n",
+        "<a href=\"https://colab.research.google.com/github/redis-developer/redis-ai-resources/blob/main/python-recipes/agents/00_langgraph_redis_agentic_rag.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>\n",
+        "\n",
+        "This notebook demonstrates the implementation of a Retrieval Agent using LangGraph and LangChain components. It showcases a flexible question-answering system that combines document retrieval with language model generation. The system uses an LLM with access to a retriever tool, making decisions about when to retrieve information from an index. Redis is utilized as a vector store for efficient document retrieval and embedding storage. Key features include adaptive query rewriting, document relevance assessment, and multi-step processing. The notebook illustrates how LangGraph can be used to create a sophisticated workflow for handling complex queries, integrating retrieval, reasoning, and generation capabilities in a single system.\n",
+        "\n",
+        "[Retrieval Agents](https://python.langchain.com/docs/tutorials/qa_chat_history/#agents) are useful when we want to make decisions about whether to retrieve from an index.\n",
+        "\n",
+        "To implement a retrieval agent, we simple need to give an LLM access to a retriever tool.\n",
```

Comment:
```suggestion
        "To implement a retrieval agent, we simply need to give an LLM access to a retriever tool.\n",
```

Reply:
Getting "Invalid email for web commit. " applying on the desktop instead